### PR TITLE
Improve sine fitting by estimating the phase

### PR DESCRIPTION
**Pull request by peendebak**
* Estimate phase in sine fit
* Add regression test

### CODE REVIEW

File: src/tests/unittests/algorithms/test_fitting.py [hunk 1]
```diff
@@ -17,30 +17,30 @@ def test_fit_sine(self):
...
                            0.56, 0.76999998, 0.88999999, 0.93000001, 0.99000001,
                            1., 0.88999999, 0.86000001, 0.74000001, 0.38])
 
-        y_data = np.array([0., 0.33069396, 0.66138792, 0.99208188, 1.32277584,
+        x_data = np.array([0., 0.33069396, 0.66138792, 0.99208188, 1.32277584,
                            1.6534698, 1.98416376, 2.31485772, 2.64555168, 2.97624564,
                            3.3069396, 3.63763356, 3.96832752, 4.29902172, 4.62971544,
                            4.96040964, 5.29110336, 5.62179756, 5.95249128, 6.28318548])
         fit_parameters, results = fit_sine(x_data, y_data)
         results['fitted_parameter_dictionary']['phase']
```
Comment (QFer) [inline src/tests/unittests/algorithms/test_fitting.py:39]: Line is not finished yet or superfluous

**Review by QFer — CHANGES_REQUESTED** [review]

File: src/tests/unittests/algorithms/test_fitting.py (hunk 1 above)
Comment (peendebak) [reply src/tests/unittests/algorithms/test_fitting.py:39]: > Line is not finished yet or superfluous

@QFer I don't understand your remark. Can you elaborate?

Comment (QFer) [reply src/tests/unittests/algorithms/test_fitting.py:39]: @peendebak The code line at line number 39 has no effect
I thought there were 2 possibilities: 1. you want to adjust it and create a test with it after all or 2. you delete the line.

Comment (peendebak) [reply src/tests/unittests/algorithms/test_fitting.py]: @QFer Thanks! I removed the line

**Review by QFer — APPROVED** [review]

**Pull request merged**